src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1393,17 +1397,31 @@ private void displayMessagesAndResult(boolean isSigning) {
     }
 
     private String verifyWithWeak(String alg, Set<CryptoPrimitive> primitiveSet,
-        boolean tsa, JarConstraintsParameters jcp) {
+        boolean tsa, JarConstraintsParameters jcp, AlgorithmParameters algParams) {
 
         try {
             JAR_DISABLED_CHECK.permits(alg, jcp, false);
         } catch (CertPathValidatorException e) {
             disabledAlgFound = true;
             return String.format(rb.getString("with.disabled"), alg);
         }
+        if (algParams != null) {
+            try {
+                JAR_DISABLED_CHECK.permits(algParams, jcp);
+            } catch (CertPathValidatorException e) {
+                disabledAlgFound = true;
+                switch (algParams.getAlgorithm().toUpperCase(Locale.ENGLISH)) {
+                    case "RSASSA-PSS":
+                        return String.format(rb.getString("with.algparams.disabled"),
+                                "RSASSA-PSS", algParams);
+                    default:
+                        return String.format(rb.getString("with.disabled"), algParams);
```

Comment:
There is no need to differentiate RSASSA-PSS and other algorithms. Just use the exact same format as you defined in `Resources.java`. This makes sure if one day another type of algorithm parameters fail the check, we will see the algorithm name and full description of the parameters as well. If the description is not clear, it's the problem of its `toString()` implementation.

Reply:
Removed code that differentiates.